ext/hash/hash.c

```diff
@@ -750,26 +750,26 @@ PHP_FUNCTION(hash_pbkdf2)
 
 	ops = php_hash_fetch_ops(algo);
 	if (!ops) {
-		zend_throw_error(NULL, "Unknown hashing algorithm: %s", ZSTR_VAL(algo));
+		zend_argument_value_error(1, "must be a valid hashing algorithm");
```

Comment:
maybe mix this and the next message together to "must be a valid cryptographic hashing algorithm"?

Reply:
Yeah, that would make sense to me.